modules/ping-protection/events/messageCreate.js

```diff
@@ -0,0 +1,113 @@
+const { 
+    addPing, 
+    getPingCountInWindow, 
+    executeAction, 
+    sendPingWarning 
+} = require('../ping-protection');
+const { localize } = require('../../../src/functions/localize');
+// Messages handler
+module.exports.run = async function (client, message) {
+    if (!client.botReadyAt) return;
+    if (!message.guild) return;
+    if (message.author.bot) return;
+    if (message.guild.id !== client.config.guildID) return;
+
+    const config = client.configurations['ping-protection']['configuration'];
+    const storageConfig = client.configurations['ping-protection']['storage'];
+    const moderationRules = client.configurations['ping-protection']['moderation'];
+    
+    if (!config || !moderationRules || !Array.isArray(moderationRules) || moderationRules.length === 0) return;
+
+    const rule1 = moderationRules[0]; 
+
+    // Checks for ignored channels and roles
+    if (config.ignoredChannels.includes(message.channel.id)) return;
+    if (message.member.roles.cache.some(role => config.ignoredRoles.includes(role.id))) return;
+
+    // Detects pings
+    const pingedProtectedRole = message.mentions.roles.some(role => config.protectedRoles.includes(role.id));
+    const pingedProtectedUser = message.mentions.users.some(user => config.protectedUsers.includes(user.id));
+    if (!pingedProtectedRole && !pingedProtectedUser) return;
+    
+    // Identifies target
+    const targetUser = message.mentions.users.find(u => config.protectedUsers.includes(u.id));
+    const targetRole = message.mentions.roles.find(r => config.protectedRoles.includes(r.id));
+    const target = targetUser || targetRole;
+    const targetName = target.tag || target.name || target.id;
+    
+    // Checks if ping history logging is enabled
+    if (!storageConfig || !storageConfig.enablePingHistory) {
```

Comment:
```suggestion
    if (!!storageConfig.enablePingHistory) {
```
correct value ensured by core